charmhelpers/contrib/network/ovs/ovsdb.py

```diff
@@ -157,23 +157,59 @@ def __init__(self, tool, table):
             self._tool = tool
             self._table = table
 
-        def _find_tbl(self, condition=None):
-            """Run and parse output of OVSDB `find` command.
-
-            :param condition: An optional RFC 7047 5.1 match condition
-            :type condition: Optional[str]
-            :returns: Dictionary with data
-            :rtype: Dict[str, any]
+        def _deserialize_ovsdb(self, data):
+            """Deserialize OVSDB RFC7047 section 5.1 data.
+
+            :param data: Multidimensional list where first row contains RFC7047
+                         type information
+            :type data: List[str,any]
+            :returns: Deserialized data.
+            :rtype: any
             """
             # When using json formatted output to OVS commands Internal OVSDB
             # notation may occur that require further deserializing.
             # Reference: https://tools.ietf.org/html/rfc7047#section-5.1
             ovs_type_cb_map = {
                 'uuid': uuid.UUID,
-                # FIXME sets also appear to sometimes contain type/value tuples
+                # NOTE: OVSDB sets have overloaded type
+                # see special handling below
                 'set': list,
                 'map': dict,
             }
+            if data[0] == 'set':
+                # special handling for set
+                #
+                # it is either a list of strings or a list of typed lists.
+                # taste first element to see which it is
+                for el in data[1]:
+                    # NOTE: We lock this handling down to the `uuid` type as
+                    # that is the only one we have a practical example of.
+                    # We could potentially just handle this generally based on
+                    # the types listed in `ovs_type_cb_map` but let's open for
+                    # that as soon as we have a concrete example to validate on
+                    if isinstance(
+                            el, list) and len(el) and el[0] == 'uuid':
+                        decoded_set = []
+                        for el in data[1]:
+                            decoded_set.append(self._deserialize_ovsdb(el))
+                        return(decoded_set)
+                    else:
+                        # fall back to normal processing below
+                        break
+                    # NOT REACHED
```

Comment:
Couldn't this just be "break".  i.e. lost the else clause.  It's a little confusing as AFAICT only the first el will be looked at in data[1] and then it will either return or break.